packages/playground/website/src/lib/tracking.ts

```diff
@@ -1,23 +1,69 @@
+import { StepDefinition } from '@wp-playground/blueprints';
+import { logger } from '@php-wasm/logger';
+
 /**
  * Declare the global window.gtag function
  */
 declare global {
-    interface Window { gtag: any; }
+	interface Window {
+		gtag: any;
+	}
 }
 
 /**
  * Google Analytics event names
  */
-type GAEvent = 'load' | 'step';
+type GAEvent = 'load' | 'step' | 'install' | 'error';
 
 /**
  * Log a tracking event to Google Analytics
  * @param GAEvent The event name
  * @param Object Event data
  */
-export const logTrackingEvent = (event: GAEvent, data?: {[key: string]: string}) => {
-    if (typeof window === 'undefined' || !window.gtag) {
-        return;
-    }
-    window.gtag('event', event, data);
-}
+export const logTrackingEvent = (
+	event: GAEvent,
+	data?: { [key: string]: string }
+) => {
+	try {
+		if (typeof window === 'undefined' || !window.gtag) {
+			return;
+		}
+		window.gtag('event', event, data);
+	} catch (error) {
+		logger.warn('Failed to log tracking event', event, data, error);
+	}
+};
+
+/**
+ * Log Blueprint step events
+ * @param step The Blueprint step
+ */
+export const logBlueprintStepEvent = (step: StepDefinition) => {
+	/**
+	 * Log the names of provided Blueprint's steps.
+	 * Only the names (e.g. "runPhp" or "login") are logged. Step options like
+	 * code, password, URLs are never sent anywhere.
+	 */
+	logTrackingEvent('step', { step: step.step });
+
+	if (step.step === 'installPlugin' && (step as any).pluginData.slug) {
+		logTrackingEvent('install', {
```

Comment:
What's the advantage of having a single `install` event with either a `plugin` or `theme` prop vs having separate `installPlugin` and `installTheme` events?

Reply:
None, it will probably be easier to build reports with separate events.
I switched to two events [beeb585](https://github.com/WordPress/wordpress-playground/pull/1984/commits/beeb585ec54e5ebd9c99a4cb752fe264967fb939)